### PR TITLE
Feature: Add color parameter to give_goal shortcode

### PR DESCRIPTION
**Pull request by rickalday**
Resolves # https://feedback.givewp.com/documentation/p/website-feedback-givewp-goal-shortcode

## Description

Adds **color** parameter to the Give Goal Shortcode.

## Affects

None

## Visuals

Screen Shot 2023-02-27 at 11 27 40 a m

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: templates/shortcode-goal.php [hunk 1]
```diff
@@ -32,7 +32,7 @@
 
 $goal_progress_stats = give_goal_progress_stats( $form );
 $goal_format         = $goal_progress_stats['format'];
-$color               = give_get_meta( $form_id, '_give_goal_color', true );
+$color               = sanitize_hex_color( $args['color'] ) ?? give_get_meta( $form_id, '_give_goal_color', true );
```
Comment (JasonTheAdams) [inline templates/shortcode-goal.php:35]: This is a little fragile. The order of operation here is that the `sanitize_hex_color()` function will _always_ be called, even if `$args['color']` is empty. Given the default of an empty string, then an empty string will be returned, and that would actually pass in a null coalescing operator comparison. Consider:
```php
echo null ?? 'hi'; // 'hi'
echo $array['not_set'] ?? 'hi'; // 'hi'
echo '' ?? 'hi': // ''
```
I recommend changing this to:
```php
$color = empty($args['color']) ? give_get_meta( $form_id, '_give_goal_color', true ) : sanitize_hex_color( $args['color'] );
```
This way the meta value is used if the parameter isn't set or is an [empty](https://www.php.net/manual/en/function.empty.php) value, and the sanitization method is only called if we're confident a value is actually set.

**Review by JasonTheAdams — CHANGES_REQUESTED** [review]
Thanks for contributing, @rickalday! I provided some feedback. Let me know if you have any questions. 😄

File: templates/shortcode-goal.php (hunk 1 above)
Comment (JasonTheAdams) [reply templates/shortcode-goal.php:35]: @rickalday This is still waiting for you. 😄

Comment (rickalday) [reply templates/shortcode-goal.php]: @JasonTheAdams I've committed the changes.

Comment (JoshuaHungDinh) [reply templates/shortcode-goal.php]: @rickalday Looks good. It's been tested and working.

**Review by JoshuaHungDinh — APPROVED** [review]

**Pull request merged**